test/unit/ui/app/actions.spec.js

```diff
@@ -574,12 +735,100 @@ describe('Actions', () => {
 
   })
 
+  describe('#signTypedMsg', () => {
+    let signTypedMsgSpy, messages, typedMessages, msgId
+
+    const msgParamsV3 = {
+      from: '0x0DCD5D886577d5081B0c52e242Ef29E70Be3E7bc',
+      data: JSON.stringify({
+        'types': {
+          'EIP712Domain': [
+            {'name': 'name', 'type': 'string'},
+            {'name': 'version', 'type': 'string'},
+            {'name': 'chainId', 'type': 'uint256'},
+            {'name': 'verifyingContract', 'type': 'address'},
+          ],
+          'Person': [
+            {'name': 'name', 'type': 'string'},
+            {'name': 'wallet', 'type': 'address'},
+          ],
+          'Mail': [
+            {'name': 'from', 'type': 'Person'},
+            {'name': 'to', 'type': 'Person'},
+            {'name': 'contents', 'type': 'string'},
+          ],
+        },
+        'primaryType': 'Mail',
+        'domain': {
+          'name': 'Ether Mainl',
+          'version': '1',
+          'chainId': 1,
+          'verifyingContract': '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC',
+        },
+        'message': {
+          'from': {
+            'name': 'Cow',
+            'wallet': '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+          },
+          'to': {
+            'name': 'Bob',
+            'wallet': '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+          },
+          'contents': 'Hello, Bob!',
+        },
+      }),
+    }
+
+    beforeEach(() => {
+      metamaskController.newUnsignedTypedMessage(msgParamsV3, 'V3')
+      messages = metamaskController.typedMessageManager.getUnapprovedMsgs()
+      typedMessages = metamaskController.typedMessageManager.messages
+      msgId = Object.keys(messages)[0]
+      typedMessages[0].msgParams.metamaskId = parseInt(msgId)
+    })
+
+    afterEach(() => {
+      signTypedMsgSpy.restore()
+    })
+
+    it('calls signTypedMsg in background with no error', () => {
+      const store = mockStore()
+      signTypedMsgSpy = sinon.spy(background, 'signTypedMessage')
+
+      return store.dispatch(actions.signTypedMsg(msgParamsV3))
+        .then(() => {
+          assert(signTypedMsgSpy.calledOnce)
+        })
+    })
+
+    it('returns expected actions with error', (done) => {
+      const store = mockStore()
+      const expectedActions = [
+        { type: 'SHOW_LOADING_INDICATION', value: undefined },
+        { type: 'UPDATE_METAMASK_STATE', value: undefined },
+        { type: 'HIDE_LOADING_INDICATION' },
+        { type: 'DISPLAY_WARNING', value: 'error' },
+      ]
+
+      signTypedMsgSpy = sinon.stub(background, 'signTypedMessage')
+      signTypedMsgSpy.callsFake((_, callback) => {
+        callback(new Error('error'))
+      })
+
+      store.dispatch(actions.signTypedMsg())
+        .catch(() => {
+          assert.deepEqual(store.getActions(), expectedActions)
+          done()
```

Comment:
This `done` is only called if an exception is thrown. If an exception is not thrown, the test never completes instead of failing.

It would probably be better just to use `async` instead of a `done` function.